addons/web_editor/static/src/js/wysiwyg/widgets/media.js

```diff
@@ -872,6 +873,39 @@ var ImageWidget = FileWidget.extend({
                 .addClass("o_we_attachment_selected");
         });
     },
+    /**
+     * @override
+     */
+    _getAttachmentsDomain(needle) {
+        const domain = this._super(...arguments);
+
+        // Optimized images (meaning they are related to an `original_id`) can
+        // only be shown in debug mode as the toggler to make those images
+        // appear is hidden when not in debug mode.
+        // There is thus no point to fetch those optimized images outside debug
+        // mode. Worst, it leads to bugs: it might fetch only optimized images
+        // when clicking on "load more" which will looks like it's bugged as no
+        // images will appear on screen (they all will be hidden).
+        if (!config.isDebug()) {
+            const subDomain = [false];
+
+            // Particular exception: if the edited image is an optimized
+            // image, we need to fetch it too so it's displayed as the
+            // selected image when opening the media dialog.
+            // We might get a few more optimized image than necessary if the
+            // original image has multiple optimized images but it's not a
+            // big deal.
+            const originalId = this.$media[0].dataset.originalId;
+            if (originalId) {
+                subDomain.push(originalId);
+            }
+
+            // TODO: use `domain.OR` in 15.0
+            domain.unshift(['original_id', 'in', subDomain]);
```

Comment:
Micro optimization but: if you can use `push`, use it instead of `unshift`. `push` just adds an element to the list, `unshift` reassigns the indexes of all existing elements.